pkg/meta/internal/crypto/aes.go

```diff
@@ -0,0 +1,123 @@
+package crypto
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"io"
+)
+
+// KeySize represents valid AES key sizes
+type KeySize int
+
+const (
+	KeySize128 KeySize = 16 // AES-128
+	KeySize192 KeySize = 24 // AES-192
+	KeySize256 KeySize = 32 // AES-256 (recommended)
+)
+
+// IsValid returns true if the key size is valid for AES
+func (ks KeySize) IsValid() bool {
+	switch ks {
+	case KeySize128, KeySize192, KeySize256:
+		return true
+	default:
+		return false
+	}
+}
+
+var ErrShortCipherText = errors.New("ciphertext too short")
+var ErrNoEncryptionKey = errors.New("encryption key is required")
+var ErrInvalidKeySize = errors.New("invalid key size: must be 16, 24, or 32 bytes")
+
+// NewKey generates a random AES key of the specified size.
+// If no size is provided, it defaults to KeySize256 (32 bytes).
+// Returns an error if the specified size is invalid or if key generation fails.
+func NewKey(size ...KeySize) ([]byte, error) {
```

Comment:
What about `NewKey()` and `NewKeyWithSize(size)`, so you avoid the variadic that gives you weird cases like `NewKeyWithSize(16, 32, 64)`. It also avoids an array allocation

Reply:
Also the name .... `GenerateKey`  is much more explicit about what's going on

Reply:
✔️